### ISSUE TITLE
Error    SA1636    The file header copyright text should match the copyright text from the settings.

### ISSUE THREAD

**Issue by sandeep-bhaskar** [DotNetAnalyzers/StyleCopAnalyzers#3271]
I am facing below issue. Even I added the company name in the settings.
Error    SA1636    The file header copyright text should match the copyright text from the settings.

**In my class file I added header like this** 
// <copyright file="RoleController.cs" company="Sandbox">
// Copyright (c) . All rights reserved.
// </copyright>

In my settings file 
{
  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
  "settings": {
    "documentationRules": {
      "documentInterfaces": true,
      "documentInternalElements": false,
      "companyName": "Sandbox",
      "copyrightText": "Copyright (c) . All rights reserved.",
      "xmlHeader": true,
      "fileNamingConvention": "metadata"
    },
    "layoutRules": {
      "newlineAtEndOfFile": "require"
    },
    "orderingRules": {
      "systemUsingDirectivesFirst": "true",
      "usingDirectivesPlacement": "outsideNamespace"
    }
  }
}

**Comment by sharwell** [comment]
What change does the code fix suggest be made?

**Comment by RoRoche** [comment]
Did you properly add the `stylecop.json` to the `.csproj` as follows?

```xml
  <ItemGroup>
    <AdditionalFiles Include="./stylecop.json" />
  </ItemGroup>
```

**Comment by smaillet** [comment]
As of StyleCop.Analyzers Version: 1.2.0.556 I am getting conflicting results on this and have to disable it outright.

1) Without any copyright I get: SA1633 (Good, that's what it should do)
2) I then run the "Add File Header" fixer for that and it generates a correct header based on my settings in StyleCop.Josn (Also GOOD!)
3) When I build, I get SA13636 saying the generated header doesn't match!
    1) This is ONLY detected at build time, so there is no IDE suggestion/fix.
    2) Even clicking on the error, it only opens the file and places the caret in the correct location of the copyright header but does NOT provide a suggestion/fix.
    3) I am left with no idea what the problem is and have to disable SA1636 all up as it is behaving like the boy that cried wolf...
    4) If I manually tweek the header to something other than what is in the stylecop.json I do get an error in the IDE (SA1636) and it offers to fix it. But if I then build the source - it generates the error again...

Basically, the "fixer" for SA1633 does NOT agree with the analyzer used for a build and the analyzer as run during the build doesn't agree with its own fix for SA1636 (And isn't reporting  what it thinks is wrong so there's no action to take to provide a clue.

>Note: in case it matters this is with a Multi-targeting project.

**Comment by bjornhellander** [comment]
@smaillet Interesting case. Can you provide some code that reproduces this behavior? What Visual Studio (or other IDE) version are you using?
